### ISSUE TITLE
Configurable audit logging output sink

### ISSUE THREAD

**Issue by embik** [kubermatic/kubermatic#8770]
### What would you like to be added? / User Story

Audit logs, if enabled, are currently scooped up by `fluent-bit` and written to stdout in that `fluent-bit` container as part of the kube-apiserver Pod.

Given the sensitivity and security implications of audit log contents, it might be desired to not write the logs to stdout (and pull them into the MLA stack, if enabled on the Seed cluster), but to write them to a configurable sink, e.g. Splunk or CloudWatch or similar (since security teams in big orgs tend to have separate logging systems). `fluent-bit` provides a list of outputs (https://docs.fluentbit.io/manual/pipeline/outputs) that we could make configurable for admins, so audit logs are processed separately.

This should probably be on the `Seed` resource and enforced down to the user clusters, as security requirements for audit logging are likely a high-level decision. Enforcement of setting up audit logging on a user cluster should stay a separate thing. If no central setting is provided via the `Seed`, it should be allowed to configure this just for the cluster in question.

### Acceptance criteria

- `Cluster` resources gain a new field `auditLog.output` (or something else? API design tbd) that configures the `fluent-bit` output for the `kube-apiserver` sidecar.
- `Seed` resources gain a new field `auditLogOutput` (or `auditLog.output`? API design tbd) that can override what is configured in a cluster and enforce the output sink settings.

**Comment by csengerszabo** [comment]
Lgtm besides one question: shouldn't we allow setting different outputs in each user clusters? I mean the current stdout `fluent-bit` containers present in each user clusters, right?

**Comment by embik** [comment]
> Lgtm besides one question: shouldn't we allow setting different outputs in each user clusters? I mean the current stdout `fluent-bit` containers present in each user clusters, right? @embik

That's correct. My reasoning for having this at the `Seed` is that in some companies, security is managed by a dedicated IT Security team that might set up rules for IT and engineering teams to play by, security-wise.

Since security standards are usually mandatory, KKP admins having the ability to enforce settings like audit log output to force compliance with them is IMHO a must-have for this feature.

But it should probably also be configurable for individual user clusters and only be overridden by higher-level settings if those are enabled. Let me rephrase the ticket a bit.
